FlowCrypt/Core/Core.swift

```diff
@@ -227,45 +227,55 @@ final class Core: KeyDecrypter, KeyParser, CoreComposeMessageType {
     }
 
     // MARK: Private calls
-    private func call(_ endpoint: String, jsonDict: [String: Any?]?, data: Data?) throws -> RawRes {
-        try call(endpoint, jsonData: try JSONSerialization.data(withJSONObject: jsonDict ?? [String: String]()), data: data ?? Data())
+    private func call(_ endpoint: String, jsonDict: [String: Any?]?, data: Data?) async throws -> RawRes {
+        return try await call(endpoint, jsonData: try JSONSerialization.data(withJSONObject: jsonDict ?? [String: String]()), data: data ?? Data())
     }
 
-    private func call(_ endpoint: String, jsonEncodable: Encodable, data: Data) throws -> RawRes {
-        try call(endpoint, jsonData: try jsonEncodable.toJsonData(), data: data)
+    private func call(_ endpoint: String, jsonEncodable: Encodable, data: Data) async throws -> RawRes {
+        return try await call(endpoint, jsonData: try jsonEncodable.toJsonData(), data: data)
     }
 
-    private func call(_ endpoint: String, jsonData: Data, data: Data) throws -> RawRes {
-        try blockUntilReadyOrThrow()
-        cb_last_value = nil
-        jsEndpointListener!.call(withArguments: [endpoint, String(data: jsonData, encoding: .utf8)!, Array<UInt8>(data), cb_catcher!])
-        guard
-            let resJsonData = cb_last_value?.0.data(using: .utf8),
-            let rawResponse = cb_last_value?.1
-        else {
-            throw CoreError.format("JavaScript callback response not available")
-        }
-
-        let error = try? resJsonData.decodeJson(as: CoreRes.Error.self)
-        if let error = error {
-            let errMsg = "------ js err -------\nCore \(endpoint):\n\(error.error.message)\n\(error.error.stack ?? "no stack")\n------- end js err -----"
-            logger.logError(errMsg)
-
-            throw CoreError(coreError: error)
+    private func call(_ endpoint: String, jsonData: Data, data: Data) async throws -> RawRes {
+        try await sleepUntilReadyOrThrow()
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<RawRes, Error>) in
```

Comment:
Since you want to have all `Core` class methods to be synchronised, you could use `actor` keyword. So you can remove queue and `withCheckedThrowingContinuation`

Reply:
Just adding actor to the class will do that?

Reply:
I just tried and needs some further refactor - I'll try again once I can at least build the app again.

Reply:
I'll file it for another issue later